src/services/azureFunctionsService.ts

```diff
@@ -94,9 +90,13 @@ export class AzureFunctionsService implements mssql.IAzureFunctionsService {
 			folderPath: projectFile
 		});
 
+		// because of an AF extension API issue, we have to get the newly created file by adding
+		// a watcher: https://github.com/microsoft/vscode-azurefunctions/issues/2908
+		const newFilePromise = azureFunctionUtils.waitForNewFunctionFile(projectFile);
```

Comment:
Oh right, I remember why we set this up now. We have to start the file watcher before we finish the create flow because otherwise if the file is created before we set up the file watcher then we'll run into a race condition where we might never get the notification the file was created.

So what we'll have to do is set up the file watcher promise before the create flow, and then as soon as that's done set up another promise with the timeout and use Promise.race to catch the first one.